lib/vector-core/src/metrics/handle.rs

```diff
@@ -131,13 +136,12 @@ impl Histogram {
     }
 
     pub(crate) fn record(&self, value: f64) {
-        for (bound, bucket) in self.buckets.iter() {
-            if value <= *bound {
-                bucket.fetch_add(1, Ordering::Relaxed);
-                break;
-            }
-        }
-
+        // The buckets are all powers of two, so calculate the ceiling of the log_2 of the value,
+        // and offset and limit it appropriately.
+        let bucket = ((value.max(Self::MIN_BUCKET).log2().ceil()) as isize - Self::MIN_BUCKET_EXP)
```

Comment:
Not really, because `value.log2()` blows up for values <= 0, which happen in tests. The `.max` there forces the value into a valid range. I'll add commentary when I break the calculation out into its own function.